packages/scripts/src/components/front-end/payments/paypal/buttons.ts

```diff
@@ -0,0 +1,334 @@
+// PayPal SDK integration for Freeform
+interface PayPalButtons {
+  render(element: Element): void;
+}
+
+type PayPalButtonsConfig = {
+  style?: Record<string, unknown>;
+  createOrder: () => Promise<string> | string;
+  onApprove: (data: { orderID: string }) => Promise<void> | void;
+  onError?: (err: unknown) => void;
+  onCancel?: (_data: unknown) => void;
+};
+
+interface PayPalSDK {
+  Buttons: (config: PayPalButtonsConfig) => PayPalButtons;
+}
+
+// Removed global declaration; use window.paypal safely instead
+
+type Config = {
+  required: boolean;
+  integration: string;
+  clientId: string;
+  sandbox: boolean;
+  currency?: string;
+  finalizeOnSubmit?: boolean;
+};
+
+const Q = {
+  root: '[data-freeform-paypal-buttons]',
+  orderInput: '[data-freeform-paypal-order]',
+};
+
+const ENDPOINTS = {
+  orders: '/freeform/payments/paypal/orders',
+} as const;
+
+const DEBUG = (window as Window & { FREEFORM_DEBUG?: boolean }).FREEFORM_DEBUG === true;
+
+function logError(...args: unknown[]) {
+  if (DEBUG) {
+    console.error('PayPal:', ...args);
+  }
+}
+
+function showNotice(root: Element, text: string) {
+  const container = root.closest('[data-field-container="payPalPayment"]') || root.parentElement;
+  if (!container) return;
+  let notice = container.querySelector('[data-ff-paypal-notice]') as HTMLElement | null;
+  if (!notice) {
+    notice = document.createElement('div');
+    notice.setAttribute('data-ff-paypal-notice', 'true');
+    notice.style.marginTop = '8px';
+    notice.style.color = '#0a7';
+    notice.style.fontSize = '0.95em';
+    container.appendChild(notice);
+  }
+  notice.textContent = text;
+}
+
+function parseConfig(el: Element): Config | null {
+  const attr = el.getAttribute('data-config');
+  if (!attr) return null;
+  try {
+    return JSON.parse(attr);
+  } catch {
+    return null;
+  }
+}
+
+type Values = Record<string, string | number | boolean | File>;
```

Comment:
Later on you specify the record value type again manually. This can be avoided by defining a type for it here.
Also since it's using `FormData` you can type it as `FormDataEntryValue`.

```suggestion
type Values = Record<string, FormDataEntryValue>;
```